test/com/google/javascript/jscomp/PolymerPassTest.java

```diff
@@ -3681,6 +3689,498 @@ public void testExportsUniqueMethodsFromLegacyElementAndBehaviors() {
+            "      },",
+            "      thingToDo: Function,",
+            "    });",
+            "  }",
+            "  _computePets() {}",
+            "  _computeName(user, thingToDo) {}",
+            "}",
+            "/** @type {!User} @private */",
+            "XElement.prototype.user_;",
+            "/** @type {!Array} */",
+            "XElement.prototype.pets;",
+            "/** @type {string} */",
+            "XElement.prototype.name;",
+            "/** @type {!Function} */",
+            "XElement.prototype.thingToDo;",
+            "(/** @suppress {uselessCode} */ function() {",
+            "  XElement.prototype._computePets;",
+            "  XElement.prototype._computeName;",
+            "  XElement.prototype.user_;",
+            "  XElement.prototype.thingToDo;",
+            "})()"));
+  }
+
+  @Test
+  public void testReflectionForComputedPropertyStrings2() {
+    propertyRenamingEnabled = true;
+
+    // Type checker doesn't currently understand ES6 code. Remove when it does.
+    disableTypeCheck();
+    test(
+        2,
+        lines(
+            "/** @constructor */",
+            "var User = function() {};",
+            "/** @type {string} */ User.prototype.id;",
+            "class XElement extends Polymer.Element {",
+            "  static get is() { return 'x-element'; }",
+            "  static get properties() {",
+            "    return {",
+            "      /** @type {!User} @private */",
+            "      user_: Object,",
+            "      pets: Array,",
+            "      name: {",
+            "        type: String,",
+            "        computed: '_computeName(\"user\", 12.0, user_.id, user_.*)'",
+            "      },",
+            "      thingToDo: Function,",
+            "    };",
+            "  }",
+            "  _computePets() {}",
+            "  _computeName(user, thingToDo) {}",
+            "}"),
+        lines(
+            REFLECT_OBJECT_DEF,
+            "/** @constructor */",
+            "var User = function() {};",
+            "/** @type {string} */ User.prototype.id;",
+            "class XElement extends Polymer.Element {",
+            "  /** @return {string} */",
+            "  static get is() { return 'x-element'; }",
+            "  /** @return {" + POLYMER_ELEMENT_PROP_CONFIG + "} */",
+            "  static get properties() {",
+            "    return $jscomp.reflectObject(XElement, {",
+            "      user_: Object,",
+            "      pets: Array,",
+            "      name: {",
+            "        type: String,",
+            "        computed: $jscomp.reflectProperty(",
+            "            '_computeName', /** @type {!XElement} */ ({})) + '(' +",
+            "            '\"user\"' + ',' + '12.0' + ',' + ",
+            "            $jscomp.reflectProperty('user_', /** @type {!XElement} */ ({})) + '.' + ",
+            "            $jscomp.reflectProperty('id', /** @type {!XElement} */ ({}).user_) + ',' + ",
```

Comment:
Just confirming that the compiler is able to understand the getprops in second parameter here? (i.e. that the result should correspond to `XElement.user_.id`?)

Reply:
Yes this works. It's not safe in general to do this - but since the reference is only used for type information and will not exist in runtime code, it will be safe here.